app/src/main/java/org/thoughtcrime/securesms/calls/WebRtcCallActivity.kt

```diff
@@ -328,44 +342,20 @@ class WebRtcCallActivity : PassphraseRequiredActionBarActivity() {
 
             launch {
                 viewModel.recipient.collect { latestRecipient ->
+                    binding.contactAvatar.recycle()
+
                     if (latestRecipient.recipient != null) {
-                        val publicKey = latestRecipient.recipient.address.serialize()
-                        val displayName = getUserDisplayName(publicKey)
-                        supportActionBar?.title = displayName
-                        val signalProfilePicture = latestRecipient.recipient.contactPhoto
-                        val avatar = (signalProfilePicture as? ProfileContactPhoto)?.avatarObject
-                        val sizeInPX =
-                            resources.getDimensionPixelSize(R.dimen.extra_large_profile_picture_size)
-                        binding.remoteRecipientName.text = displayName
-                        if (signalProfilePicture != null && avatar != "0" && avatar != "") {
-                            glide.clear(binding.remoteRecipient)
-                            glide.load(signalProfilePicture)
-                                .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
-                                .circleCrop()
-                                .error(
-                                    AvatarPlaceholderGenerator.generate(
-                                        this@WebRtcCallActivity,
-                                        sizeInPX,
-                                        publicKey,
-                                        displayName
-                                    )
-                                )
-                                .into(binding.remoteRecipient)
-                        } else {
-                            glide.clear(binding.remoteRecipient)
-                            glide.load(
-                                AvatarPlaceholderGenerator.generate(
-                                    this@WebRtcCallActivity,
-                                    sizeInPX,
-                                    publicKey,
-                                    displayName
-                                )
-                            )
-                                .diskCacheStrategy(DiskCacheStrategy.ALL).circleCrop()
-                                .into(binding.remoteRecipient)
+                        val contactPublicKey = latestRecipient.recipient.address.serialize()
```

Comment:
This serialize() method is comedy (not that you wrote it - just everytime I see it I want to slap myself). If you take a look all it does is lowercase the string, and it even does that using a deprecated method call (`toLowerCase`). I'd love to remove that serialize() method and just have a custom `get()` on address that lowercases it.

Reply:
Hmm yeah I never looked inside. There's definitely a lot of work we can do to clean up and centralise anything to do with getting the user's data: public key and username...